### ISSUE TITLE
HTML/Body element issues overlooked

### ISSUE THREAD

**Issue by ssimsRIT** [validator/validator#169]
The validator returns "No errors or warnings to show." for pages that have no opening body element, but do contain a closing body element and vice versa. The same is true for the html element.

For example the following will validate.

```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<title>My Example Page</title>
		<meta charset="utf-8"  />
	</head>
	<nav>
        	<a href="index.html">Home</a>
	</nav>
	</body>
```

Removing the closing </nav> tag triggers Error: End tag for  body seen, but there were unclosed elements. and Error: Unclosed element nav.

**Comment by sideshowbarker** [comment]
The `body` start tag is optional per the HTML spec; see the **Tag omission in text/html** section at https://html.spec.whatwg.org/multipage/semantics.html#the-body-element and the similar info at https://developer.mozilla.org/en/docs/Web/HTML/Element/body

And to be clear, the `body` start tag is still optional even if the `body` end tag is present. They are both optional always, regardless of whether one or the other is present.

And all the same is true for the `html` start tag and end tag.

> Removing the closing tag triggers Error: End tag for body seen, but there were unclosed elements. and Error: Unclosed element nav.

You mean removing the `nav` end tag, right? Because as far as I can see, that’s the only condition in which you’d see that message. And if you remove the `nav` end tag, it is the expected behavior that you would see that message—because the `nav` end tag is not optional (unlike the `body` end tag).

**Comment by ssimsRIT** [comment]
Thank you for pointing out these nuances to the standard. It seems odd to require opening/closing tags for everything except two major building blocks of the document. Imagine someone just learning the language that is taught every opening tag requires a closing tag, except html and body as long as these specific criteria are met.

**Comment by sideshowbarker** [comment]
> It seems odd to require opening/closing tags for everything except two major building blocks of the document. Imagine someone just learning the language that is taught every opening tag requires a closing tag, except html and body as long as these specific criteria are met.

Yeah I agree but it’s worth remembering that the parsing behavior for HTML has from almost its beginning been such that you could omit the `body` and `html` end tags and start tags. XHTML came along in the mean time and a lot of people got confused by that and thought it changed/overrode the original existing HTML rules that allow those tags to be emitted. But the HTML behavior itself actually remained exactly the same during all the XHTML years, and that’s what we have now (and XHTML is basically now just a footnote in history…).
